### DISCUSSION TITLE
HP COMPAQ SFF computers: Is it possible to control the cooling system FAN on their motherboards from the operating system?

### DISCUSSION THREAD

**Discussion by Vasya07** [Rem0o/FanControl.Releases#1755] [Q&A]
Hello everybody! Unfortunately (maybe not) I realized that I can't control the fan speed of my PC (HP COMPAQ 6300 Pro SFF) from the operating system (Windows 10 x64 version 21H2). I have a Fan Control V155 and I can only control the fan on my graphics card (GT1030) using this program.

Actually, I'm looking for a program that can help me save (maybe) my processor (Intel Core i5-3470) from overheating, because my cooling system always starts to cool the processor only at a temperature of 80 degrees Celsius or more, and I think my processor does not smile when it has such a high temperature :(

I have already tried a bunch of programs (SpeedFan, RLECViewer Hwinfo64, Notebook Fan Control), but not one of them could help me cope with my problem. I know that maybe my PC will make noise like an airplane turbine, but the most important thing is that I will be able to protect my processor from overheating.

I will look forward to the answer to my question, and even if it turns out that I am losers and that it is impossible to control the cooling system fan on my PC at all. :(, then I will at least put an end to the question that has been tormenting me for 2 years (see the question in the title of the topic).

Thanks to the author for such a good program. I am sure that a little more and this program will become the best in the category "PC cooling system fan control"!

Thanks to everyone in advance who answers this question! 
Thank you all!

**Comment by cdkone** [comment]
Hi,
I am definitely not an expert, but I have an old Gigabyte motherboard that was difficult to get control of the fans. I found Argus Monitor [https://www.argusmonitor.com/](https://www.argusmonitor.com/) It does have a cost, but it works well with my board.

**Comment by Vasya07** [comment]
Thanks for the answer, but unfortunately, Argus Monitor also does not allow me to control the CPU fan speed, only the graphics card. As a last resort, I'll buy myself a rheobas, but this is an extreme case

**Comment by Vasya07** [comment]
I tried once again to create a configuration for Notebook Fancontrol (NBFC), but without success. It seems to me that HP has some proprietary way of controlling fans on the CPU. Perhaps if I dig into the BIOS code, then it will be possible to learn how to control the CPU fan speed through the system? But since I am an ordinary PC user and do not understand the BIOS codes at all, it may not make sense to develop this topic further (see the question in the title of the topic) :( 

That 's all for now

**Reply by Vasya07** [reply]
But I will wait and maybe some answers on this topic will come.

**Reply by Vasya07** [reply]
I just now opened the JSON file (configuration file of the fan control program) and saw there that the fan of my processor is simply invisible to this program. Thus, it seems that there is no support for HP computers in this program, at least not yet (I hope the author will add it soon) or is there, but I just don't understand something?

**Reply by Rem0o** [reply]
These pre-built were not made with user-controlled fan control in mind. They most likely have a set logic in place, and that's it, no customizability.

**Reply by Vasya07** [reply]
Okay, thanks for the answer. If I understand you correctly, you want to say that the only way to control the fans on HP COMPAQ SFF computers is to create a modified BIOS in which to prescribe at what temperatures the processor cooling system should increase the fan speed, and at what temperatures the processor cooling system should reduce the fan speed, and that's it (well, and the second way is to buy a reobas)?

The most interesting thing, however, is that on my motherboard the CPU cooling system fan and even the power supply fan have a 4-pin connector, but apparently for some reason no program I know can control any fan from the operating system (in the BIOS it is possible to set the minimum rotation speed fan - tab "Power", option "Thermal") - see attachments.

I hope to hear some positive news from the author and from the entire community of this program and thank you again for your response



![CHA_FAN on mainboard](https://user-images.githubusercontent.com/104777757/236755450-1eca52d4-9e6e-4728-96fc-df9a985632c2.jpeg)

![FAN connector on the CPU cooler](https://user-images.githubusercontent.com/104777757/236755456-762eb03c-222c-4ec5-856b-767feb9bc733.jpeg)

![CHA_FAN 4-pin connector on mainboard and CPU cooler](https://user-images.githubusercontent.com/104777757/236755448-2c9ccf43-5d08-43ff-8a9b-7c02ced83365.jpeg)

![Only CHA_FAN 4-pin connecor](https://user-images.githubusercontent.com/104777757/236755446-0c881ea6-9cc7-4458-a1b6-57cd0478968f.jpeg)

![BIOS Power Thermal and F1 button (about this function)](https://user-images.githubusercontent.com/104777757/236755424-dec81fcd-2630-4683-80b6-5edc77566364.jpeg)

**Comment by Vasya07** [comment]
I found a very interesting thing in the BIOS of my PC. 

I have a BIOS from AMI, so I found a program to edit their BIOS. I converted my original BIOS into a "rom." file, and then opened it in this program and found there 3 (!) additional tabs that are not visible to the average PC user in the BIOS. My plan is to open these tabs (I've already done that) and sew the modified BIOS into my PC's motherboard (I haven't done that yet). I have already conducted an experiment and did the same with my old ASUS A42F laptop, which also has a BIOS from AMI. I managed to open these tabs and now I can control the cooling system of my laptop directly from BIOS (modified) and do many different interesting things - in short, God mode in BIOS.

Now I will no longer need any programs to control the CPU fan because I can do it all right in BIOS (modified).

Yes it looks like I now sysadmin of my PC or something like this :)
